### DISCUSSION TITLE
MRTK3 tutorial - Integrate Azure Custom Vision

### DISCUSSION THREAD

**Discussion by dldnwlsgd** [microsoft/MixedRealityToolkit-Unity#11787] [Q&A]
In the process of integrating custom vision, I created a custom vision account and entered the key and ID into the 'ObjectDetectionManager'. Afterwards, when I press the Train button in Unity Play mode, the following error appears:

```c#
Exception: PermissionDenied
MRTK.Tutorials.AzureCloudServices.Scripts.Managers.ObjectDetectionManager.TrainProject () (at Assets/MRTK.Tutorials.AzureCloudServices/Scripts/Managers/ObjectDetectionManager.cs:113)
MRTK.Tutorials.AzureCloudServices.Scripts.Controller.ComputerVisionController.StartModelTraining () (at Assets/MRTK.Tutorials.AzureCloudServices/Scripts/Controller/ComputerVisionController.cs:199)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) (at <9d9536d9127f4a489d989c7a566aee1c>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at <120b8d04add741329ccd415c000fb666>:0)
UnityEngine.UnitySynchronizationContext.Exec () (at <120b8d04add741329ccd415c000fb666>:0)
UnityEngine.UnitySynchronizationContext.ExecuteTasks () (at <120b8d04add741329ccd415c000fb666>:0)
``` 

Can anyone solve this problem??

**Comment by marlenaklein-msft** [comment]
Hi @dldnwlsgd, were you able to resolve this? 

There should be two different key/endpoint pairs: one for 'YourResourceName' and one for the corresponding prediction resource 'YourResourceName-prediction'. In the [tutorial](https://learn.microsoft.com/en-us/training/modules/azure-cloud-services-tutorials/7-exercise-integrate-azure-custom-vision), the **Retrieve Azure setting credentials** and **Retrieve project setting credentials** sections should describe how to assign both of these. 

(also for questions related to MRTK3, please post discussions in the new repository (https://github.com/orgs/MixedRealityToolkit/discussions) :)

**Reply by dldnwlsgd** [reply]
Hello. Thank you for your reply.

However, I couldn't solve this problem. I confirmed that the photo was uploaded to the Custom Vision website, but since then, when I press the "Train" button in Unity, an error continues to appear. Can you help me with what problem I'm having?

`Exception: PermissionDenied
MRTK.Tutorials.AzureCloudServices.Scripts.Managers.ObjectDetectionManager.TrainProject () (at Assets/MRTK.Tutorials.AzureCloudServices/Scripts/Managers/ObjectDetectionManager.cs:113)
MRTK.Tutorials.AzureCloudServices.Scripts.Controller.ComputerVisionController.StartModelTraining () (at Assets/MRTK.Tutorials.AzureCloudServices/Scripts/Controller/ComputerVisionController.cs:199)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) (at <9d9536d9127f4a489d989c7a566aee1c>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at <120b8d04add741329ccd415c000fb666>:0)
UnityEngine.UnitySynchronizationContext.Exec () (at <120b8d04add741329ccd415c000fb666>:0)
UnityEngine.UnitySynchronizationContext.ExecuteTasks () (at <120b8d04add741329ccd415c000fb666>:0)
`
